### DISCUSSION TITLE
XAD integration with a large codebase

### DISCUSSION THREAD

**Discussion by Manushanker** [auto-differentiation/XAD#86] [General]
I integrated the XAD library with an in-house electromagnetic solver to compute the Jacobian and Hessian matrixes. After integration the  memory and computation time required to determine the gradients is quite high! I tried different examples but faced a similar issue. I want to use XAD as it computes the first and second order derivatives with a single functional call. Further, not many open-source libraries have the capability to compute the Hessian using the forward mode. May I know if it is possible for you to look at my large codebase integration and resolve this issue. If required, I will share the code with all the input files. 

It would be excellent if XAD works seamlessly with my code, as I am facing extreme limitations with other AD libraries such as CppAD, CoDiPACK, Auto_diff. Your help will be highly appreciated!

Thanks,

Manushanker

**Comment by auto-differentiation-dev** [comment]
Thanks for reaching out Manushanker.  

The maintainers and contributors of XAD are focused on improving the tool itself and fixing related issues as they are flagged. If you could narrow down the problem into a small reproducer, we can have a look and see what the issue is. If it turns out to be a bug, we can add your reproducer as a test and fix it. 

Otherwise, if you feel there isn't any bug/issue with XAD itself and your question relates to a specific bespoke integration issue with a larger application, then we typically refer such queries to Xcelerit as they have a commercial offering covering consultancy. You can find details on their website: https://www.xcelerit.com/#contact
